pkg/addonmanager/controllers/addoninstall/controller.go

```diff
@@ -87,6 +87,23 @@ func (c *addonInstallController) sync(ctx context.Context, syncCtx factory.SyncC
 
 		switch addon.GetAgentAddonOptions().InstallStrategy.Type {
 		case agent.InstallAll:
+			return c.applyAddon(ctx, addonName, clusterName, addon.GetAgentAddonOptions().InstallStrategy.InstallNamespace)
+		case agent.InstallByLabel:
+			labelSelector := addon.GetAgentAddonOptions().InstallStrategy.LabelSelector
+			if labelSelector == nil {
+				return nil
```

Comment:
Is it worth logging a warning or if there is a condition to say you've asked for the labelSelector, but you didn't provide one?